benchmarks/benchmark.py

```diff
@@ -0,0 +1,754 @@
+                            var_name,
+                            make_arg_function,
+                            arg_type,
+                            value,
+                        )
+                    )
+                    if opencl == "base":
+                        setup += "  auto {} = stan::math::to_matrix_cl({});\n".format(
+                            var_name + "_cl", var_name
+                        )
+                        var_name += "_cl"
+                    elif varmat == "base" and arg_overload == "Rev":
+                        setup += "  auto {} = stan::math::to_var_value({});\n".format(
+                            var_name + "_varmat", var_name
+                        )
+                        var_name += "_varmat"
+                else:
+                    var_conversions += (
+                        "    {} {} = stan::test::{}<{}>({}, state.range(0));\n".format(
+                            arg_type,
+                            var_name,
+                            make_arg_function,
+                            arg_type,
+                            value,
+                        )
+                    )
+                    if opencl == "base":
+                        var_conversions += (
+                            "    auto {} = stan::math::to_matrix_cl({});\n".format(
+                                var_name + "_cl", var_name
+                            )
+                        )
+                        var_name += "_cl"
+                    elif varmat == "base" and arg_overload == "Rev":
+                        var_conversions += (
+                            "    auto {} = stan::math::to_var_value({});\n".format(
+                                var_name + "_varmat", var_name
+                            )
+                        )
+                        var_name += "_varmat"
+                if opencl == "copy" and stan_arg not in ("int", "real"):
+                    code += "stan::math::to_matrix_cl({}), ".format(var_name)
+                elif (
+                        varmat == "copy"
+                        and stan_arg not in ("int", "real")
+                        and arg_overload == "Rev"
+                ):
+                    code += "stan::math::to_var_value({}), ".format(var_name)
+                else:
+                    code += var_name + ", "
+            if opencl == "base":
+                var_conversions += "  stan::math::opencl_context.queue().finish();\n"
+            code = code[:-2] + "));\n"
+            if "Rev" in arg_overloads:
+                code += "    stan::math::grad();\n"
+            result += BENCHMARK_TEMPLATE.format(
+                benchmark_name=benchmark_name,
+                setup=setup,
+                var_conversions=var_conversions,
+                code=code,
+                multi=multiplier,
+                max_size=max_size,
+            )
+    cpp_filepath = WORKING_FOLDER + cpp_filename
```

Comment:
```suggestion
    cpp_filepath = os.path.join(WORKING_FOLDER, cpp_filename)
```

It is safer to use os.path if WORKING_FOLDER is changed by the user